### PR TITLE
[CN-891] fix mutation detection for CR update

### PR DESCRIPTION
**Pull request by SeriyBg**
Update the CRD when mutated in the reconciler

### CODE REVIEW

File: controllers/hazelcast/mutate/mutate.go [hunk 1]
```diff
@@ -10,6 +12,10 @@ func HazelcastSpec(h *hazelcastv1alpha1.Hazelcast) (mutated bool) {
 		h.Spec.Repository = n.HazelcastEERepo
 		mutated = true
 	}
+	oldHz := &h
 	h.Default()
+	if !reflect.DeepEqual(oldHz, &h) {
```
Comment (semihbkgr) [inline controllers/hazelcast/mutate/mutate.go:17]: return statement can be simplified

**Review by semihbkgr — CHANGES_REQUESTED** [review]
It is just small suggestion.

File: controllers/hazelcast/mutate/mutate.go (hunk 1 above)
Comment (SeriyBg) [reply controllers/hazelcast/mutate/mutate.go:17]: `mutated` variable might be changed above. IMO, such simplification will make the code less readable

Comment (semihbkgr) [reply controllers/hazelcast/mutate/mutate.go:17]: I meant something like this
```go
oldHz := &h
h.Default()
if h.Spec.GetLicenseKeySecretName() != "" && h.Spec.Repository == n.HazelcastRepo {
		h.Spec.Repository = n.HazelcastEERepo
		return true
}
return !reflect.DeepEqual(oldHz, &h)
```
not big deal, It is still good anyway 👍

**Review by semihbkgr — APPROVED** [review]
LGTM

**Review by hasancelik — APPROVED** [review]

**Pull request merged**